cpp/src/utilities/prefetch.cpp

```diff
@@ -34,13 +34,18 @@ prefetch_config& prefetch_config::instance()
 
 bool prefetch_config::get(std::string_view key)
 {
-  // Default to not prefetching
+  std::shared_lock<std::shared_mutex> lock(config_mtx);
   if (config_values.find(key.data()) == config_values.end()) {
-    return (config_values[key.data()] = false);
+    return false;  // default to not prefetching
   }
-  return config_values[key.data()];
+  return config_values.at(key.data());
```

Comment:
This does a double-lookup which is wasteful.  Can avoid it with something like this:
```c++
  auto const i = config_values.find(key.data());
  if (i == config_values.end()) {
    return false;
  } else {
    return i->second;
  }
```